### ISSUE TITLE
Remmina doesn't respect ssh config files...

### ISSUE THREAD

**Issue by kosilin** [FreeRDP/Remmina#235]
so is very sad...
I suppose to add one line with ssh_options_parse_config() call just after ssh session object creation.
##

**Comment by antenore** [comment]
I've added the support for ~/.ssh/config, but keep in mind that most of the ssh options are not supported by libssh , so I'm not sure it will works as you like, please test it.

I'm going to merge this in 'next'

**Comment by Hubbitus** [comment]
Thank you very much!

**Comment by antenore** [comment]
^_^

**Comment by str** [comment]
Is this working?  I have reminna  1.10 and My ForwardAgent defined in ~.ssh/config is not used.  I reported it in bug #267 which was just closed but I can't  confirm it's fixed.

**Comment by antenore** [comment]
1.2.0-rcgit.4 is the version that has been fixed, the version you are using is not supported anymore, can you follow the instructions in the wiki to update and test it?

https://github.com/FreeRDP/Remmina/wiki

**Comment by str** [comment]
@antenore  I've added the ppa and downlaoded 1.2.0-rcgit.4 but i still can't use ForwardAgent

**Comment by antenore** [comment]
@str 

The ~/.ssh/config parsing works, but libssh doesn't support all of the ssh_config options and we don't have too much power on it.

I've activated the debug window finding this disturbing message

```
[SSH] ssh_config_parse_line: Unsupported option: ForwardAgent, line: 7
```
This message come from the libssh, that is the library we use in Remmina.
We have chosen this library because it's actually the best option we have, the alternatives would be:
 1. Develop a libssh wrapper that implements part of the missing features (that's a bad idea)
 2. Develop from scratch a new library (several working days).
 3. Support the libssh team in order to help them developing what is missing.

All of the above means we should reduce the limited time we pass improving Remmina (we do it in our free time) to improve the someone else project or working full time to write our own library.

To make it short, at the moment I cannot do better.
Maybe you could try to understand if and when the libssh team they will support all the ssh_config options?

**Comment by str** [comment]
I understand it's not an easy task. I'm checking the libssh2 documentation and reading about forward agent feature in libssh.  If this is a pending feature, can we leave the task/feature-request open?

**Comment by antenore** [comment]
Sure, why not...

**Comment by RangerGH** [comment]
I'm adding to this as I'm running 1.2.0-0.8.git.b3237e8.fc23 on Fedora 23.  I created an ssh connection to myserver which uses port 2222 and an identity file which is setup in my ssh config file but if I do not enter the server as myserver:2222 in Remmina it will not connect.  It appears it's not using the ssh config file.  I can connect via command line ssh just fine.

I found Remmina while looking for the Linux equivalent of mtty.  I also use another Linux remote connection app but so far prefer Remmina.  I still have to deal with Windows machines and Remmina handles them better than the other and I'd like to have to use only one app.
